pkg/operation/botanist/component/dependencywatchdog/dependency_watchdog.go

```diff
@@ -362,6 +362,17 @@ func (d *dependencyWatchdog) clusterRoleRules() []rbacv1.PolicyRule {
 				Resources: []string{"deployments", "deployments/scale"},
 				Verbs:     []string{"get", "list", "watch", "update"},
 			},
+			{
+				APIGroups: []string{"coordination.k8s.io"},
+				Resources: []string{"leases"},
+				Verbs:     []string{"create"},
+			},
+			{
+				APIGroups:     []string{"coordination.k8s.io"},
+				ResourceNames: []string{"dependency-watchdog-probe", "dependency-watchdog-endpoint"},
+				Resources:     []string{"leases"},
+				Verbs:         []string{"get", "watch", "update"},
+			},
```

Comment:
I am not quite sure about this handling. To my understanding we have a switch case for the dependency-watchdog role. Which means that dependency-watchdog-endpoint should receive similar RBAC rules. And `dependency-watchdog-probe` should receive only RBAC rules related to the `dependency-watchdog-probe`.

```suggestion
			{
				APIGroups:     []string{"coordination.k8s.io"},
				ResourceNames: []string{"dependency-watchdog-probe"},
				Resources:     []string{"leases"},
				Verbs:         []string{"get", "watch", "update"},
			},
```

Reply:
You are right there is a separate switch for endpoint. Again an oversight on my end, thanks for catching it. 
I'll fix it, and also verify it with a test. Currently i'm facing some issues with the local setup, shall update the PR with the fix and test results before update the PR commits.

Reply:
This is fixed now with the following commit https://github.com/gardener/gardener/commit/baf409c8e937da5dc46f6572cd2be322ed330fd8. 
I've also verified with a local shoot and the endpoint controller is able to start post the change introduced with RBAC. 
Also tested the scaleDown and scaleUp scenarios for the probe and it seems to work fine.